pollux/lib/core/src/test/scala/io/iohk/atala/pollux/core/service/PresentationServiceSpec.scala

```diff
@@ -358,12 +374,192 @@ object PresentationServiceSpec extends ZIOSpecDefault with PresentationServiceSp
+            case _ => ZIO.fail(InvalidAnoncredPresentationRequest("Expecting Base64-encoded data"))
+          }
+          repo <- ZIO.service[PresentationRepository]
+          credentialsToUse =
+            AnoncredCredentialProofsV1(
+              List(
+                AnoncredCredentialProofV1(
+                  aIssueCredentialRecord.id.value,
+                  Seq("sex"),
+                  Seq("age")
+                )
+              )
+            )
+          credentialsToUseJson <- ZIO.fromEither(
+            AnoncredCredentialProofsV1.schemaSerDes.serialize(credentialsToUse)
+          )
+          _ <-
+            repo.updateAnoncredPresentationWithCredentialsToUse(
+              aRecord.id,
+              Some(AnoncredPresentationV1.version),
+              Some(credentialsToUseJson),
+              PresentationRecord.ProtocolState.RequestPending
+            )
+          issuer = createIssuer(DID("did:prism:issuer"))
+          presentation <- svc.createAnoncredPresentation(
+            aRecord.requestPresentationData.get,
+            aRecord.id,
+            issuer,
+            credentialsToUse,
+            Instant.now()
+          )
+          _ <- svc.receivePresentation(presentation)
+          serialisedPresentation <- presentation.attachments.head.data match {
+            case Base64(data) => ZIO.succeed(AnoncredPresentation(new String(JBase64.getUrlDecoder.decode(data))))
+            case _            => ZIO.fail(InvalidAnoncredPresentation("Expecting Base64-encoded data"))
+          }
+          validateRecord <-
+            svc.verifyAnoncredPresentation(
+              presentation,
+              aRecord.requestPresentationData.get,
+              aRecord.id
+            )
+        } yield {
+          val presentation1 = serialisedPresentation
+          val presentation2 =
+            AnoncredLib.createPresentation(
+              serialisedPresentationRequest,
+              Seq(
+                AnoncredCredentialRequests(AnoncredCredential(processedCredential.data), Seq("sex"), Seq("age"))
+              ),
+              Map.empty, // TO FIX
+              linkSecret.secret,
+              Map(credentialOffer.schemaId -> schema), // schemas: Map[SchemaId, SchemaDef],
+              Map(
+                credentialOffer.credDefId -> credentialDefinition.cd
+              )
+            )
+          val verifyPresentation = AnoncredLib.verifyPresentation(
+            presentation2.toOption.get, // : Presentation,
+            serialisedPresentationRequest, // : PresentationRequest,
+            Map(credentialOffer.schemaId -> schema), // schemas: Map[SchemaId, SchemaDef],
+            Map(
+              credentialOffer.credDefId -> credentialDefinition.cd
+            ), // credentialDefinitions: Map[CredentialDefinitionId, CredentialDefinition],
+          )
+          println(s"AnoncredPresentationRequest: ${serialisedPresentationRequest}")
```

Comment:
I wouldn't keep `println` in the tests.
I think, sooner or later, we need to invest some resources into making our stdout as empty as possible.

Reply:
removed